### ISSUE TITLE
Recommendation stops while in fight - Unholy DK

### ISSUE THREAD

**Issue by SirColeslaw** [Hekili/hekili#562]
1.  Check to see if you are using the latest version of the addon.  If there is a newer alpha/beta/release version, see if your problem exists with the new version.
#Hekili-v9.0.2-1.0.4-beta1#

2.  Check to see if someone else has already opened an issue report here.  Please don't submit repeat reports.
#I could not find anything#

**Describe the Bug/Issue**
It can happen that when I'm in combat the addon doesn't recommend which action to take next. Up to version Hekili-v9.0.2-1.0.0 this has always worked reliably and without problems.

The problem usually occurs after a combat duration of 30-60 seconds. As soon as i change my target the recommendations start again.

**To Reproduce**
Play a Unholy DK.
Attack a target which results in a longer fight.
Watch the recommendations.

**Expected behavior**
I would expect that until the death of the target, the recommendations would not stop.

**Screenshots**
https://ibb.co/bdXMChS

Normally I have a 5 button queue. But in the screenshot, although the fight is still going on, only one button is shown.

**WHEN SUBMITTING AN ISSUE:**

Please provide the following information.  Leaving it out may result in your ticket being ignored (because I won't have the information needed to explore your report).

1.  The information from the Issue Reporting tab ( /hekili > Issue Reporting ).  This provides your talent and gear information.  You can paste it to pastebin.com and provide a link here.

**https://pastebin.com/EFx9niCP**

2.  If reporting a disagreement with the addon's recommendations, please take a snapshot when you observe the issue and provide the snapshot.  This is not a screenshot.  A snapshot can be generated by binding the Pause feature on the Toggles tab.  When you press Pause, a snapshot of how the addon decided on its CURRENT recommendations is generated and saved to the Snapshots tab.  Copy and paste this to pastebin.com and provide the link here.
 
**https://pastebin.com/YkruewyH**

**Comment by Hekili** [comment]
Can you provide any more detail about the circumstances when the recommendations fail?  Is it multiple targets?  Anything else stand out about the circumstances?

**Comment by SirColeslaw** [comment]
First of all the answers to your questions:

My character is level 45, which is certainly not optimal, and Hekili gives a hint that it is actually made for level 50 characters, but I had no problems until version 9.0.1-1.0.0.

So I tested a little bit, and here's what I found:

On a single training dummy in Stormwind the problem does not occur, even after a longer time.

It does not occur on several training manikins in Stormwind, even after a longer time.

It does not occur on a raremob, even after a longer period of time.

With multiple targets, however, I was able to reproduce it several times now, so I suspect it only occurs in multiple target situations. I could also observe this when a longer running raremob fight was joined by other opponents, only then the error started to appear.

Here is another snapshot of such a situation where my queue runs empty.

https://pastebin.com/6XUNEhpW
